File: app/src/main/java/com/hellowo/teamfinder/ui/adapter/TagListAdapter.java
package com.hellowo.teamfinder.ui.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.hellowo.teamfinder.R;
import com.hellowo.teamfinder.model.HashTag;

import java.util.List;


public class TagListAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
    private Context context;
    private List<HashTag> mContentsList;
    private boolean isEditable;
    private AdapterInterface adapterInterface;

    public TagListAdapter(Context context, boolean isEditable, List<HashTag> mContentsList,
                          AdapterInterface adapterInterface) {
        this.context = context;
        this.mContentsList = mContentsList;
        this.isEditable = isEditable;
        this.adapterInterface = adapterInterface;
    }

    public class ItemViewHolder extends RecyclerView.ViewHolder {
        View container;
        TextView titleText;
        ImageView imageView;

        public ItemViewHolder(View v) {
            super(v);
            container = v;
            titleText = (TextView) v.findViewById(R.id.titleText);
            imageView = (ImageView) v.findViewById(R.id.imageView);
        }
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup viewGroup, int viewType) {
        View v = LayoutInflater.from(context)
                .inflate(R.layout.list_item_basic, viewGroup, false);
        return new ItemViewHolder(v);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder viewHolder, int position) {
        final HashTag hashTag = mContentsList.get(position);
        ItemViewHolder holder = (ItemViewHolder)viewHolder;
        holder.titleText.setText(hashTag.getName());
        holder.imageView.setImageResource(hashTag.getIconId());
        holder.container.setOnClickListener(v -> adapterInterface.onItemClicked(hashTag));
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    @Override
    public int getItemCount() {
        return mContentsList.size();
    }

    public interface AdapterInterface {
        void onItemClicked(HashTag hashTag);
    }
}

File: app/src/main/java/com/hellowo/teamfinder/ui/adapter/TagListAdapter.kt
package com.hellowo.teamfinder.ui.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.hellowo.teamfinder.R
import com.hellowo.teamfinder.model.HashTag
import kotlinx.android.synthetic.main.list_item_tag.view.*

class TagListAdapter(val context: Context,
                     val mContentsList: List<HashTag>,
                     val adapterInterface: (HashTag) -> Unit) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {

    inner class ViewHolder(container: View) : RecyclerView.ViewHolder(container)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int)
            = ViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_item_tag, parent, false))

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        val hashTag = mContentsList[position]
        val v = holder.itemView
        v.tagText.text = hashTag.makeTag()
        v.setOnClickListener { adapterInterface.invoke(hashTag) }
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    override fun getItemCount(): Int {
        return mContentsList.size
    }
}